.circleci/config.yml

```diff
@@ -0,0 +1,310 @@
+            pyenv versions
+            pyenv global 3.6.5
+            pyenv versions
+      - run:
+          name: Virtual Environment
+          command: |
+            python -m venv ~/code/venv
+            echo "source ~/code/venv/bin/activate" >> $BASH_ENV
+      - run:
+          name: Install Requirements
+          command: |
+            echo $(which python)
+            pip install -r requirements.txt
+      - run:
+          name: Test Requirements
+          command: |
+            echo $(which python)
+            pip install -r test_requirements.txt
+      - save-venv-cache
+
+  # Specific collections of python tests
+  test-builder:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: builder
+
+  test-cli:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: cli
+
+  test-client:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: client
+
+  test-data-provider:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: dataprovider
+
+  test-dataset:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: dataset
+
+  test-model:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: model
+
+  test-serializer:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: serializer
+
+  test-server:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: server
+
+  test-util:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: util
+
+  test-watchman:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: watchman
+
+  test-formatting:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: formatting
+
+  test-allelse:
+    executor: python-executor
+    steps:
+      - run-test:
+          for: allelse
+
+workflows:
+  version: 2.1
+
+  # Run all tests, ensure images and docs can be built
+  test:
+    jobs:
+      - make-docs
+      - install
+      - test-builder:
+          requires:
+            - install
+      - test-cli:
+          requires:
+            - install
+      - test-client:
+          requires:
+            - install
+      - test-data-provider:
+          requires:
+            - install
+      - test-dataset:
+          requires:
+            - install
+      - test-model:
+          requires:
+            - install
+      - test-serializer:
+          requires:
+            - install
+      - test-server:
+          requires:
+            - install
+      - test-util:
+          requires:
+            - install
+      - test-watchman:
+          requires:
+            - install
+      - test-formatting:
+          requires:
+            - install
+      - test-allelse:
+          requires:
+            - install
+      - make-images:
+          filters:
+            branches:
+              only: /.*/
+            tags:
+              only: /.*/
```

Comment:
Dont we want to try making images always, but only push them on tags?

Reply:
Ya it looks a bit awkward, but for any steps which has `filters` such as `push-images` any step it depends on must also specify an equal-to-or-looser equivalent filter. Slightly strange quirk.